modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectActionLocalServiceImpl.java

```diff
@@ -411,6 +417,20 @@ private void _validateParametersUnicodeProperties(
 				_objectDefinitionPersistence.fetchByPrimaryKey(
 					objectDefinitionId);
 
+			if (parametersUnicodeProperties.containsKey(
+					"objectDefinitionExternalReferenceCode")) {
+
+				String objectDefinitionExternalReferenceCode =
+					GetterUtil.getString(
+						parametersUnicodeProperties.get(
+							"objectDefinitionExternalReferenceCode"));
+
+				objectDefinition = _objectDefinitionPersistence.fetchByERC_C(
+					objectDefinitionExternalReferenceCode, companyId);
+
+				objectDefinitionId = objectDefinition.getObjectDefinitionId();
```

Comment:
As we discussed:

1. This variable "objectDefinition" can be null and will override the variable "objectDefinitionId"
2. We should prioritize the objectDefinitionExternalReferenceCode. If we find the Object Definition, we should override the objectDefinitionId.
3. If we don't find by objectDefinitionExternalReferenceCode, fetch by "objectDefinitionId".
4. We should delete objectDefinitionExternalReferenceCode from the parametersUnicodeProperties because we cannot persist this information.